content/tutorials/core-ledger/create-custom-transactors.md

```diff
@@ -0,0 +1,389 @@
+                    keylet::line(srcId, issuerId, sendMax.getCurrency()));
+                if (sleTrust &&
+                    sleTrust->isFlag(
+                        (issuerId > srcId) ? lsfHighFreeze : lsfLowFreeze))
+                {
+                    JLOG(ctx.j.warn())
+                        << "Creating a check for frozen trustline.";
+                    return tecFROZEN;
+                }
+            }
+            if (issuerId != dstId)
+            {
+                // Check if dst froze the line.
+                auto const sleTrust = ctx.view.read(
+                    keylet::line(issuerId, dstId, sendMax.getCurrency()));
+                if (sleTrust &&
+                    sleTrust->isFlag(
+                        (dstId > issuerId) ? lsfHighFreeze : lsfLowFreeze))
+                {
+                    JLOG(ctx.j.warn())
+                        << "Creating a check for destination frozen trustline.";
+                    return tecFROZEN;
+                }
+            }
+        }
+    }
+    if (hasExpired(ctx.view, ctx.tx[~sfExpiration]))
+    {
+        JLOG(ctx.j.warn()) << "Creating a check that has already expired.";
+        return tecEXPIRED;
+    }
+    return tesSUCCESS;
+}
+```
+
+
+### 3. Add a `doApply()` function.
+
+The `doApply()` function has read/write access, enabling you to modify the ledger.
+
+```c++
+CreateCheck::doApply()
+{
+    auto const sle = view().peek(keylet::account(account_));
+    if (!sle)
+        return tefINTERNAL;
+
+    // A check counts against the reserve of the issuing account, but we
+    // check the starting balance because we want to allow dipping into the
+    // reserve to pay fees.
+    {
+        STAmount const reserve{
+            view().fees().accountReserve(sle->getFieldU32(sfOwnerCount) + 1)};
+
+        if (mPriorBalance < reserve)
+            return tecINSUFFICIENT_RESERVE;
+    }
+
+    // Note that we use the value from the sequence or ticket as the
+    // Check sequence.  For more explanation see comments in SeqProxy.h.
+    std::uint32_t const seq = ctx_.tx.getSeqProxy().value();
+    Keylet const checkKeylet = keylet::check(account_, seq);
+    auto sleCheck = std::make_shared<SLE>(checkKeylet);
+
+    sleCheck->setAccountID(sfAccount, account_);
+    AccountID const dstAccountId = ctx_.tx[sfDestination];
+    sleCheck->setAccountID(sfDestination, dstAccountId);
+    sleCheck->setFieldU32(sfSequence, seq);
+    sleCheck->setFieldAmount(sfSendMax, ctx_.tx[sfSendMax]);
+    if (auto const srcTag = ctx_.tx[~sfSourceTag])
+        sleCheck->setFieldU32(sfSourceTag, *srcTag);
+    if (auto const dstTag = ctx_.tx[~sfDestinationTag])
+        sleCheck->setFieldU32(sfDestinationTag, *dstTag);
+    if (auto const invoiceId = ctx_.tx[~sfInvoiceID])
+        sleCheck->setFieldH256(sfInvoiceID, *invoiceId);
+    if (auto const expiry = ctx_.tx[~sfExpiration])
+        sleCheck->setFieldU32(sfExpiration, *expiry);
```

Comment:
This can be modified to be easier to read by using the `[]` operator (not sure how exact you want the code to be to the original version)